benchmarks/src/jmh/kotlin/benchmarks/flow/FlowFlattenMergeBenchmark.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2016-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+package benchmarks.flow
+
+import doGeomDistrWork
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.runBlocking
+import org.openjdk.jmh.annotations.*
+import java.util.concurrent.TimeUnit
+
+/**
+ * Benchmark to measure performance of [kotlinx.coroutines.flow.FlowKt.flattenMerge].
+ * In addition to that, it can be considered as a macro benchmark for the [kotlinx.coroutines.sync.Semaphore]
+ */
+@Warmup(iterations = 3, time = 1)
+@Measurement(iterations = 10, time = 1)
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@State(Scope.Benchmark)
+@Fork(1)
+open class FlowFlattenMergeBenchmark {
+
+    /**
+     * Number of flows that are merged in this benchmark. Negative number means that number of flows
+     * will be computed as -([flows] * [concurrency]), positive number will be chosen as number of flows.
+     */
+    @Param("-10", "-1", "100", "500")
+    private var flows: Int = 0
```

Comment:
This parameter is really hard to interpret.
E.g. when running the benchmark, I observe the following output:
```
FlowFlattenMergeBenchmark.flattenMerge              1       -1  thrpt   10  43.670 ± 2.254  ops/s
FlowFlattenMergeBenchmark.flattenMerge              1      100  thrpt   10  43.384 ± 0.723  ops/s
```

How an unprepared reader should interpret `-1` and `100`? It doesn't look like the dependency between `concurrency` and `flows` is really necessary here